plugins/autosuspend.koplugin/main.lua

```diff
@@ -141,34 +147,58 @@ function AutoSuspendWidget:addToMainMenu(menu_items)
                 ok_text = _("Set timeout"),
                 title_text = _("Timeout in minutes"),
                 callback = function(autosuspend_spin)
-                    G_reader_settings:saveSetting("auto_suspend_timeout_seconds", autosuspend_spin.value * 60)
-                    -- NOTE: Will only take effect after a restart, as we don't have a method to set this live...
+                    local autosuspend_timeout_seconds = autosuspend_spin.value * 60
+                    self.auto_suspend_sec = autosuspend_timeout_seconds
+                    G_reader_settings:saveSetting("auto_suspend_timeout_seconds", autosuspend_timeout_seconds)
                     UIManager:show(InfoMessage:new{
-                        text = _("This will take effect on next restart."),
+                        text = T(_("The system will automatically suspend after %1 minutes of inactivity."),
+                            string.format("%.2f", autosuspend_timeout_seconds/60)),
+                        timeout = 3,
                     })
+                    self:_unschedule()
+                    self:_start()
+                end
+            }
+            UIManager:show(autosuspend_spin)
+        end,
+    }
+    if not (Device:isKobo() or Device:isEmulator()) then return end
```

Comment:
Should use canPowerOff, not isKobo
 https://github.com/koreader/koreader/blob/07116b98982bc65a158e877e4a001a8563b0ca82/frontend/device/generic/device.lua#L46